### ISSUE TITLE
Export and import annotations (annotations imported and placed on the wrong position) 

### ISSUE THREAD

**Issue by xchrx** [stephanrauh/ngx-extended-pdf-viewer#1825]
First of all thank you for your work!

We have the case to store and restore the annotations on the pdf. if you try to restore the annotation, the anntation would created in the right size, but placed on the wrong position. It is offset by the size of the annotation.

Steps for recreating the bug:
1. temporary save the current annotation information with the method getSerializedAnnoations();
2. remove the set annotations with removeEditorAnnotations();
3. restore the temporary saved annotations with addEditorAnnoation();

There you have the bug - placed on the wrong position

**Comment by stephanrauh** [comment]
The offset is the same offset used by the copy-and-paste feature. That might be a clue.

**Comment by slimouGit** [comment]
unfortunately I ve the same issue.
For instance with following code:

const serializedAnnotations = this.pdfService.getSerializedAnnotations();
this.pdfService.removeEditorAnnotations();
this.pdfService.addEditorAnnotation(serializedAnnotations[0]);

the inserted annotation wont be placed at the original position

**Comment by PatrickZimmerer** [comment]
I have the same issue, when copy-pasting it works fine, but where can I adjust the offset? The pasted annotation is always moved to the bottom right for the exact height & width of the annotation.

**Comment by stephanrauh** [comment]
By now, I'm pretty sure that the offset is caused by me being lazy. I used the "paste" part of the copy-and-paste implementation of the editors. Unfortunately, yesterdays debugging session didn't reveal where the offset is added. Maybe you can help me? During the next two or three weeks I'm short of time, so I'd appreciate your time.

Set `[minifiedJSLibraries]="false"`, open `pdf-xx.xxx.js` in the debugger, and add a breakpoint here: https://github.com/stephanrauh/pdf.js/blob/6f55fbbdeba9d55227211c2799f1f770b95767a0/src/display/editor/tools.js#L835.

Thanks in advance,
Stephan

**Comment by stephanrauh** [comment]
Found it! I've managed to solve the issue, at least for drawings. It'll take a few days to fix it for texts and stamps, but after the first breakthrough it should be easy.

**Comment by stephanrauh** [comment]
Your bugfix has landed with version 18.0.0-beta.2.

Please note that the internal coordinates of the serialized drawings have changed. This caught me by surprise, too, and it took a while until I figured it out. The `bezier` and `path` attributes now use absolute coordinates (i.e. relative to the page). Previously, they used coordinates relative to the origin of the drawing.

**Comment by stephanrauh** [comment]
Oops. The version only fixed the ink editor. Version 18.0.0-beta.3 fixes the offset for all three editors.

The showcase often shows an offset of 2 units. I don't know where this comes from. But at least it's a predictable offset now, and chances are it's caused by the showcase and not by the library itself.

**Comment by slimouGit** [comment]
great job, thanks a lot. Annotations are placed at origin position now.
But: in case you draw multiple annotations, each on seperate page, all annotations are placed at this page its currently in view.
The padgeindex is known, but is not used for insertion

**Comment by stephanrauh** [comment]
What? Oh no! That's unexpected. As a short-term workaround, use `[page]` to navigate to the target page before adding an annotatation.

**Comment by slimouGit** [comment]
yes indeed, I allready tryed this workaround, for instance like this: 
 
```typescript

annotations.forEach(a => {
      this.scroll(a.pageIndex+1,0)
      this.pdfService.addEditorAnnotation(a);
      });
```
it works fine so far
```

**Comment by stephanrauh** [comment]
I haven't published it yet, but version 18.0.0-beta.5 is going to fix that. Most likely, I'm going to publish the version this week-end, so I'm closing the ticket now.
